### PR TITLE
Micu/mark currently applicable line in the opening hours in venue description #289

### PR DESCRIPTION
**Pull request by micuintus**

### CODE REVIEW

**Comment by jmastr** [comment]
@micuintus Don't know, if it is part of your patch, but shouldn't be the opening hours here in one line?

**Comment by micuintus** [comment]
@jmastr Good catch!

> Don't know, if it is part of your patch,

No, this was already part of #237

> but shouldn't be the opening hours here in one line?

The problem is: If it is a holiday people might get confused why not the current day was chosen. Therfore we renamed Subday to Sunday / Holdiday, but now the collapsing looks akward for that, which is why we removed it.

**Comment by jmastr** [comment]
> The problem is: If it is a holiday people might get confused why not the current day was chosen. Therfore we renamed Subday to Sunday / Holdiday, but now the collapsing looks akward for that, which is why we removed it.
We can leave it for now, but this solution isn't ideal either...

**Comment by micuintus** [comment]
> > The problem is: If it is a holiday people might get confused why not the current day was chosen. Therfore we renamed Sunday to Sunday / Holiday, but now the collapsing looks akward for that, which is why we removed it.
> 
> We can leave it for now, but this solution isn't ideal either...

Agreed, its not. But: It was not introduced within the scope of this ticket :D
On the other hand, one could argue that marking the open line mitigates the issue why we renamed  Sunday to Sunday / Holiday, a bit. But I'd say lets leave this for another ticket.

File: src/OpeningHoursAlgorithms.h [hunk 1]
```diff
@@ -30,27 +30,30 @@
...
-                          const unsigned until)
+                          const unsigned until,
+                          const int todayIndex)
 {
+    QVariantMap result;
+
     if (from == until)
     {
-        return openingHours[from].toMap();
+        result = openingHours[from].toMap();
```
Comment (jmastr) [inline src/OpeningHoursAlgorithms.h]: OpeningHoursAlgorithms.h:40: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

File: src/OpeningHoursAlgorithms.h [hunk 2]
```diff
@@ -30,27 +30,30 @@
...
     }
     else
     {
-        QVariantMap result;
-
         const auto& fromDay  = openingHours[from]. toMap()["day"].toString();
```
Comment (jmastr) [inline src/OpeningHoursAlgorithms.h]: OpeningHoursAlgorithms.h:44: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

File: src/OpeningHoursAlgorithms.h [hunk 3]
```diff
@@ -30,27 +30,30 @@
...
         const auto& untilDay = openingHours[until].toMap()["day"].toString();
```
Comment (jmastr) [inline src/OpeningHoursAlgorithms.h]: OpeningHoursAlgorithms.h:45: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

File: src/OpeningHoursAlgorithms.h [hunk 4]
```diff
@@ -30,27 +30,30 @@
...
 
-        result["day"]   = fromDay + " - " + untilDay;
-        result["hours"] = openingHours[from].toMap()["hours"];
-
-        return result;
+        result["day"]      = fromDay + " - " + untilDay;
+        result["hours"]    = openingHours[from].toMap()["hours"];
```
Comment (jmastr) [inline src/OpeningHoursAlgorithms.h]: OpeningHoursAlgorithms.h:48: warning: implicit conversion changes signedness: 'const unsigned int' to 'int'

File: src/OpeningHoursAlgorithms.h [hunk 5]
```diff
@@ -30,27 +30,30 @@
...
     }
+
+    result["current"]  = from <= todayIndex && todayIndex <= until;
```
Comment (jmastr) [inline src/OpeningHoursAlgorithms.h]: OpeningHoursAlgorithms.h:51: warning: comparison of integers of different signs: 'const unsigned int' and 'const int'
OpeningHoursAlgorithms.h:51: warning: comparison of integers of different signs: 'const int' and 'const unsigned int'

File: src/VenueModel.cpp [hunk 6]
```diff
@@ -282,15 +244,28 @@ QVariant VenueModel::data(const QModelIndex &index, int role) const
...
+            return QVariant::Invalid;
+        }
 
-    return QStandardItemModel::data(index, role);
+        return condenseOpeningHours(openingHoursVar.toList(), m_currendDayIndex);
+    }
+    default:
+        return QStandardItemModel::data(index, role);
+    }
 }
```
Comment (jmastr) [inline src/VenueModel.cpp]: VenueModel.cpp:270: warning: control may reach end of non-void function [-Wreturn-type]

**Comment by jmastr** [comment]
Let's tackle that in a follow-up ticket, since the PR did not introduce the warning and the PRs is big enough already.

**Pull request merged**